### PR TITLE
Add initial support for sticky positioning for non-legacy layout

### LINKED ISSUE #30570: Debug assertion failures in components/layout_2020/fragment_tree/box_fragment.rs:261
(#30567, #30509)

```
% ( set -- 2020/filtered-test-wpt.*.json; for i; do < $i jq -r '.[] | select(.stack != null) | .stack' | rg -o --pcre2 '(?<=, at )[^()]+(?=[)]$)'; done | sort -u | while read -r site; do echo ">>> $site"; for i; do < $i jq -r '.[] | select(.stack != null) | select(.stack | contains("'"$site"'")) | .stack'; done | rg ', at |^assertion failed:|^  left:|^ right:' | sort -u; done )
>>> components/layout_2020/fragment_tree/box_fragment.rs:261
Got unknown position. (thread Layout(1,1), at components/layout_2020/fragment_tree/box_fragment.rs:261)
```

* layout 2020
    * /css/cssom/getComputedStyle-insets-sticky-container-for-abspos.html
    * /css/cssom/getComputedStyle-insets-sticky.html
    * /css/cssom/getComputedStyle-sticky-pos-percent.html
    * /css/css-position/inheritance.html

### PR DESCRIPTION
**Pull request by mrobinson**
Many tests still fail for a variety of reasons. One of the primary ones
is that CSSOM currently does not return correct values for elements
positioned by sticky nodes. This requires changes to WebRender to work
properly.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #30570.
- [x] There are tests for these changes

### CODE REVIEW

**Comment by mrobinson** [comment]
I'm converting this back to a draft for now. We need to fix expectations and also two assertion failures uncovered by requiring a display list for answering resolved style queries.

**Comment by mrobinson** [comment]
Okay. This one is ready for review.

**Review by mukilan — APPROVED** [review]

File: tests/wpt/meta/css/css-position/sticky/position-sticky-table-td-left.html.ini [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [position-sticky-table-td-left.html]
-  expected: FAIL
+  expected: TIMEOUT
```
Comment (mukilan) [inline tests/wpt/meta/css/css-position/sticky/position-sticky-table-td-left.html.ini:2]: Are these tests that have TIMEOUT  just crashes that don't exit servo?  Or do they indicate that implementation is slow on CI?

Comment (mrobinson) [reply tests/wpt/meta/css/css-position/sticky/position-sticky-table-td-left.html.ini:2]: Nope. I think these are failing for some other reason. I do see this error when running them:

```
 [2023-11-28T17:07:01Z ERROR script::dom::bindings::error] Error at http://web-platform.test:8000/css/css-position/resources/ref-rectangle.js:14:3 document.fonts is undefined
```

So I suspect it's related to the lack of `documents.fonts` support.

**Pull request merged**